superdesk/upload.py

```diff
@@ -55,6 +58,50 @@ def get_upload_as_data_uri(media_id):
     raise SuperdeskApiError.notFoundError("File not found on media storage.")
 
 
+@bp.route("/upload/config-file", methods=["POST", "OPTIONS"])
+@blueprint_auth()
+def upload_config_file():
+    if request.method == "OPTIONS":
+        # return headers to avoid CORS problems
+        response = make_response()
+        response.headers.add("Access-Control-Allow-Origin", "*")
+        response.headers.add("Access-Control-Allow-Headers", "*")
+        response.headers.add("Access-Control-Allow-Methods", "POST")
+        return response
+
+    if not current_user_has_privilege("vocabularies"):
```

Comment:
imo it would be good to get the resource via request param so we can pick the service and privilege based on that.
but we should have some whitelist of resources which can be updated like this.

Reply:
For `VOCABULARIES MANAGEMENT` the privilege name is `vocabularies`, so here it is the privilege name, not the resource. however, we know that for vocabularies, the resource and privilege names are the same but it does not work for all privileges and resources.
Like we have one privilege called `UNPUBLISH` but do not have resources with the same name.

Reply:
yep but we can use the resource to figure out its privileges right?

Reply:
Okay got it, we will send resource via request param and then find out its privileges and then based on that allow the user to do further processes so that we can reuse that functionality for other resources also right?

Reply:
exactly